### ISSUE TITLE
Naming and Pointing - how to use the data collected within the whole training stage

### ISSUE THREAD

**Issue by afansi** [maximecb/baby-ai-game#2]

**Comment by afansi** [comment]
I am thinking that the part of model for naming and pointing may be used in two ways:
1. During the naming and pointing procedure:
We use the data collected, that we continuously saved in a kind of internal memory, to train the classifier. The latter receives as input a pair (text, image). Here, maybe we can train the classifier ONLY, not the other part of the whole RL model.
2. When the naming an pointing procedure is over:
The classifier is used in  "inference mode" to provide additional signal to the RL model

What do you think?

**Comment by maximecb** [comment]
A link to Simon's notes regarding our last meeting: https://docs.google.com/document/d/1ibZp8eyUFQRPq_DfZ0qXtOXqv_AZ21AyQhU5Qwpe8Is/edit

In @simon555's diagram, it looks like the classifier feeds into the historical state RNN, which I believe is in agreement with what you said.

You're right that the classifier doesn't have to be trained at the same time as we train RL. Probably, it should be trained every couple of episodes. Sort of an interleaved mode of training, where we do some RL, then some pointing and naming, then some RL training again.

**Comment by simon555** [comment]
@afansi You are right, no need to involve the RL part of the model when we will train the naming and pointing module.

As @maximecb pointed, the output of the classifier feed into the RNN but we will only need to use the visual and text modules in order to train the classifier. 
Once trained, the whole connections between models will be used.

**Comment by maximecb** [comment]
Just a quick update to say that I added some basic support in the UI for gathering pointing and naming data: https://github.com/maximecb/baby-ai-game/blob/82b7079f2051a5a819a150ec33116fd7a1c2e811/main.py#L198

This is in the master branch. It generates both positive and negative examples of the object being clicked by randomly placing the agent around the position that was clicked and described.

There is currently no model or training code attached to this.

**Comment by maximecb** [comment]
I added a trivial classifier-type model in the `maxime-pointing` branch: https://github.com/maximecb/baby-ai-game/blob/maxime-pointing/model/training.py
